src/useForm.tsx

```diff
@@ -90,13 +94,37 @@ function createFields<T>(
           // This avoids issues related to stale state when multiple setValue functions are
           // invoked within the same render cycle -- e.g.
           // const onClick = () => { date.day.setValue(...); date.day.setValue(...) }:
-          setValue: (updatedValue: any) => {
-            setState(currentState =>
-              produce(currentState, updatedState => {
-                set(updatedState, [...path, 'value'], updatedValue)
-                set(updatedState, [...path, 'touched'], true)
-              })
-            )
+          setValue: <
+            U extends
+              | SetValueOptions
+              | EmptySetValueOptions = EmptySetValueOptions
+          >(
+            updatedValue: any,
+            options: SetValueOptions | undefined = undefined
```

Comment:
What's the purpose of `| undefined` here? I would have expected that we could just make `options` optional with `options?: SetValueOptions`.

Also, why do we need to explicitly provide a default of `undefined`? I would have expected it to be undefined if we made it optional and then didn't specify a value.

Reply:
Ah, great question.

I think this was an oversight on my part where I originally had a different method signature here and forgot to come back and clean it up. 

Pretty sure we can have this just be `options?: SetValueOptions`

Reply:
Confirmed: https://github.com/ginger-io/react-use-form/pull/9/commits/ec247bb0d5d536a851ce437efa850a5cf6477983